.github/workflows/main.yml

```diff
@@ -0,0 +1,51 @@
+name: Static analysis
+run-name: GitHub Actions for clang-tidy
+
+on: pull_request
+
+jobs:
+  clang-tidy:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - name: Fetch base branch
+        run: |
+          git remote add upstream "https://github.com/${{ github.event.pull_request.base.repo.full_name }}"
+          git fetch --no-tags --no-recurse-submodules upstream "${{ github.event.pull_request.base.ref }}"
+          git submodule update --init --recursive
+
+      - name: Install clang-tidy
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y clang clang-tidy libldap2-dev curl libcurl4-openssl-dev bison
+
+      - name: Prepare compile_commands.json
+        run: |
+          cmake -B build -DCMAKE_INSTALL_PREFIX=../install -DWITH_DEBUG=1 -DDOWNLOAD_BOOST=1 -DWITH_BOOST=my_boost \
+          -DWITH_ZLIB=bundled -DWITH_SSL=system -DWITH_AUTHENTICATION_LDAP=0 -DWITH_LZ4=bundled \
+          -WITH_KEYRING_VAULT=ON -DWITH_ROCKSDB=0 -DCMAKE_C_COMPILER=clang-14 -DCMAKE_CXX_COMPILER=clang++-14 \
```

Comment:
Above you install `clang clang-tidy` so it's better to use `clang` and `clang++` here.
With `ubuntu-22.04` `clang` probably leads to `clang-14` but it may not be the case in the future if we will change `ubuntu-22.04` to something else.

Reply:
Another option is to install `clang++-17 llvm-17-dev clang-tidy-17` from http://apt.llvm.org
as we do it at https://github.com/percona/percona-server/blob/8.0/azure-pipelines.yml#L396-L400

Reply:
Will update clang-14 and clang++-14 to clang and clang++ and let it take care of the versions.